src/utils/log.c

```diff
@@ -351,10 +345,16 @@
         *txt++ = 't';
         break;
       default:
+        // check if value is a valid printable ASCII char
+        // this also confirms that we can safely cast unsigned data to signed
+        // char
         if (data[i] >= 32 && data[i] <= 126) {
-          *txt++ = data[i];
+          *txt++ = (char)data[i];
         } else {
-          txt += snprintf(txt, end - txt, "\\x%02x", data[i]);
+          // guaranteed to be positive and to have 4 chars left, as otherwise
+          // loop will break
+          size_t max_chars_to_print = (size_t)(end - txt);
+          txt += snprintf(txt, max_chars_to_print, "\\x%02x", data[i]);
```

Comment:
This code scanning alert is a false positive, since we check whether `txt` is over the end at https://github.com/nqminds/edgesec/blob/6b1a286a6e4f64e9bd4a0c008f6430b72413ee87/src/utils/log.c#L325